### PR TITLE
Generalize the way we handle runtime components when writing diagnostics

### PR DESCRIPTION
**Pull request by atmyers**
Right now, there is some hard coding about what comps to expect based on whether we are running with QED, ionization, etc...

This changes this to just ask the source `pc` what components and names to use.

### CODE REVIEW

**Comment by atmyers** [comment]
@roelof-groenewald I think something like this is what we need.

**Comment by roelof-groenewald** [comment]
Great! Thanks for looking into this. I'll merge this branch into mine to confirm that it fixes my problem in PR #2115 .

File: Source/Diagnostics/WarpXOpenPMD.cpp [hunk 1]
```diff
@@ -433,26 +434,29 @@ WarpXOpenPMDPlot::WriteOpenPMDParticles (const amrex::Vector<ParticleDiag>& part
...
-        real_names.push_back("opticalDepthQSR");
-        tmp.AddRealComp(false);
-    }
-#endif
+    // add runtime real comps to tmp
+    for (int ic = 0; ic < pc->NumRuntimeRealComps(); ++ic) { tmp.AddRealComp(false); }
+
+    // get the names of the real comps
+    real_names.resize(pc->NumRealComps());
+    auto runtime_rnames = pc->getParticleRuntimeComps();
```
Comment (ax3l) [inline Source/Diagnostics/WarpXOpenPMD.cpp:476]: Here we need to be careful with `_` in names, as we use them currently as separators for vector/tensor components.

In the current form this re-introduces #1777

Comment (atmyers) [reply Source/Diagnostics/WarpXOpenPMD.cpp:476]: I have fixed this by converting these names to camelCase, as they are in `development`. 

We need a more systematic way of handling scalar versus vector component names, but for now I think we should merge this as it is blocking a few different PRs.

Comment (ax3l) [reply Source/Diagnostics/WarpXOpenPMD.cpp:476]: Clarified:
> converting these names

is only the runtime added attributes, so momentum et al (added above) that are added by default are unaffected.

**Review by ax3l — APPROVED** [review]
Thank you! :+1:

**Pull request merged**